teps/0008-support-knative-service-for-triggers-eventlistener-pod.md

```diff
@@ -0,0 +1,286 @@
+---
+title: support-knative-service-for-triggers-eventlistener-pod
+authors:
+  - "@savitaashture"
+  - "@vdemeester"
+creation-date: 2020-07-28
+last-updated: 2020-07-28
+status: proposed
+---
+
+# TEP-0008: Support Knative Service for Triggers Eventlistener Pod
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+- [Design Details](#design-details)
+- [Advantages](#advantages)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This proposal helps users to deploy triggers eventlistener as Knative Service along with the existing Kubernetes Deployments. 
+And even users can bring their own CRD with a defined contract for spec and status.
+Original Google Doc proposal visible to members of tekton-dev@: [design doc](https://docs.google.com/document/d/1GtCfpzgGFPt224A7xNE5sjN8REytytmNL8E9oXAk4Uc/edit?usp=sharing)
+
+## Motivation
+
+Triggers in conjunction with Pipeline enables the creation of full-fledged CI/CD systems.
+Triggers EventListener helps to process incoming HTTP based events with JSON payloads,
+So there may be chances that the EventListener should be capable to handle more requests without dropping any of those
+and should automatically scales up and down based on requests and should not use any resources during its idle state,
+So to achieve all those triggers should be more flexible to support different approaches.
+
+### Goals
+
+One goal is to get the benefits of serverless features.
+
+Second goal of this proposal is to provide flexibility to the user to bring their 
+own CRD with a specified spec contract to deploy triggers eventlistener pod.
+
+### Non-Goals
+
+Deploying and maintaining the dependencies is not the goal of this proposal.
+Ex: Deploying and maintaining of Knative will not be part of the Triggers.
+
+## Proposal
+
+Allow triggers to support Knative Service to deploy eventlistener pod 
+along with the existing kubernetes deployment in order to support the functionality of serverless.
+With the support Knative Service triggers get the serverless features without any additional configuration.
+
+### User Stories
+
+A user who is concerned about autoscale to not lose a single request,
+Resource usage when not using triggers eventlistener then they can use Knative service in order to solve the problem. 
+
+## Design Details
+
+Kubernetes Deployment, Knative Service both have PodSpec as a common so usage of
```

Comment:
Should this say "as a common sub-field" or something? I think a word is missing.